compiler/rustc_resolve/src/late.rs

```diff
@@ -360,6 +360,7 @@ enum LifetimeBinderKind {
     Function,
     Closure,
     ImplBlock,
+    ImplAssocType,
```

Comment:
Could you add a comment to the `Item` variant explaining which items it covers? Is it only trait assoc types now?